supabase/migrations/20230817110845_access_token.sql

```diff
@@ -0,0 +1,174 @@
+create table app.access_tokens(
+    id uuid primary key default gen_random_uuid(),
+    user_id uuid not null references auth.users(id),
+    token_hash bytea not null,
+    token_name text not null check (length(token_name) <= 64),
+    created_at timestamptz not null default now(),
+
+    unique (user_id, token_name)
+);
+
+grant insert
+    (id, user_id, token_hash, token_name)
+    on app.access_tokens
+    to authenticated;
+
+grant delete
+    on app.access_tokens
+    to authenticated;
+
+alter table app.access_tokens enable row level security;
+
+create policy access_tokens_select_policy
+    on app.access_tokens
+    as permissive
+    for select
+    to public
+    using ( auth.uid() = user_id );
+
+create policy access_tokens_insert_policy
+    on app.access_tokens
+    as permissive
+    for insert
+    to authenticated
+    with check ( auth.uid() = user_id );
+
+create policy access_tokens_delete_policy
+    on app.access_tokens
+    as permissive
+    for delete
+    to authenticated
+    using ( auth.uid() = user_id );
+
+create or replace function public.new_access_token(
+    token_name text
+)
+    returns text
+    language plpgsql
+    strict
+as $$
+declare
+    user_id uuid = auth.uid();
+    account app.accounts = account from app.accounts account where id = user_id;
+    token bytea = gen_random_bytes(16);
+    token_hash bytea = pgsodium.crypto_pwhash_str(token);
+    token_text text = encode(token, 'hex');
+begin
+    begin
+        insert into app.access_tokens(user_id, token_hash, token_name)
+        values (account.id, token_hash, token_name);
+    exception when unique_violation then
+        raise exception 'Token with name `%s` already exists', token_name;
+    end;
+
+    return replace(user_id::text, '-', '') || token_text;
+end;
+$$;
+
+create type app.access_token_struct as (
+    id uuid,
+    token_name text,
+    created_at timestamptz
+);
+
+create or replace function public.get_access_tokens()
+    returns setof app.access_token_struct
+    language plpgsql
+    strict
+as $$
+declare
+    user_id uuid = auth.uid();
+    account app.accounts = account from app.accounts account where id = user_id;
+    token bytea = gen_random_bytes(16);
+    token_hash bytea = pgsodium.crypto_pwhash_str(token);
+    token_text text = encode(token, 'hex');
```

Comment:
It's often useful to encode some identifying information in the token's prefix. For example: `dbdev_<token>`.

Also what you may want to store in the database is the token's prefix (first 4-5 chars) to help people identify it quickly. It may also be a bit more compact to encode as Base64-URL or something.

Reply:
Implemented this, tokens now look like this:

<img width="898" alt="image" src="https://github.com/supabase/dbdev/assets/1666073/780fb6f8-afcc-411d-9bff-342dfc73deab">

Still a bit long due to `token_id` being uuids. I'm trying to convert them to an integer which should make the tokens smaller.